stdlib/2.7/__builtin__.pyi

```diff
@@ -529,7 +529,7 @@ class dict(MutableMapping[_KT, _VT], Generic[_KT, _VT]):
     @overload
     def __init__(self, map: Mapping[_KT, _VT], **kwargs: _VT) -> None: ...
     @overload
-    def __init__(self, iterable: Iterable[Tuple[_KT, _VT]], **kwargs: _VT) -> None: ...
+    def __init__(self, iterable: Iterable[Union[Tuple[_KT, _VT], Sequence[_VT]]], **kwargs: _VT) -> None: ...
```

Comment:
Yeah, this is more permissive than the real type. The situation is very similar to the one with `**kwargs`, as commented above. I believe this example would work much the same way as the one you mentioned:
```
d = dict(x='y')  # type: Dict[int, str]
```

For keyword arguments, what we really want to say is that those are allowed only if `_KT` is `str`. And for general sequences, we really want to say that those are allowed only if `_KT` equals `_VT`. But as you say, we don't have a way to represent that in the PEP 484 type system. I think this is the right thing to do in the circumstances (I sat with Tyler the other day looking at this problem), but it deserves a comment.

The union is a little awkward -- I'd have preferred two separate overloads -- but mypy didn't respond well to that, probably related to the fact that tuples are sequences.